pkg/util/codec/codec.go

```diff
@@ -1586,6 +1587,20 @@ func ConvertByCollationStr(str string, tp *types.FieldType) string {
 	return string(hack.String(collator.Key(str)))
 }
 
+// Hash64 is for datum hash64 calculation.
+func Hash64(h base.Hasher, d *types.Datum) {
+	// let h.cache to receive datum hash value, which is potentially expendable.
```

Comment:
passing hasher into every encoding logic of datum field means we need to mutate many codec functions, which is dirty.

here we just choose to let hasher ingest datum.HashCode() bytes to make it.